Composer/packages/client/src/recoilModel/dispatchers/utils/project.ts

```diff
@@ -418,10 +469,35 @@ export const initBotState = async (callbackHelpers: CallbackInterface, data: any
 
   set(filePersistenceState(projectId), new FilePersistence(projectId));
   set(undoHistoryState(projectId), new UndoHistory(projectId));
-
-  // async repair bot assets, add missing lg templates
-  repairBotProject(callbackHelpers, { projectId, botFiles });
-
+  set(projectIndexingState(projectId), true);
+  Promise.all([
+    lgWorker.parseAll(projectId, lgFiles).then((result) => {
+      set(lgFilesSelectorFamily(projectId), (prevFiles) => {
+        return prevFiles.map((item) => {
+          const file = (result as LgFile[]).find((file) => file.id === item.id);
+          return file && item.rawData ? file : item;
+        });
+      });
+    }),
+    luWorker.parseAll(luFiles, mergedSettings.luFeatures).then((result) => {
+      set(luFilesSelectorFamily(projectId), (prevFiles) => {
+        return prevFiles.map((item) => {
+          const file = (result as LuFile[]).find((file) => file.id === item.id);
+          return file && item.rawData ? file : item;
+        });
+      });
+    }),
+    qnaWorker.parseAll(qnaFiles).then((result) => {
+      set(qnaFilesSelectorFamily(projectId), (prevFiles) => {
+        return prevFiles.map((item) => {
+          const file = (result as QnAFile[]).find((file) => file.id === item.id);
+          return file && item.rawData ? file : item;
+        });
+      });
+    }),
+  ]).then(() => {
```

Comment:
I wonder if this would be easier to follow if the promises were just `worker.parseAll(...)` and then we updated state in the `then`. I would even suggest using `await` as well.

`const [luResult, lgResult, qnaResult] = await Promsie.all(luWorker, lgWorker, qnaWorker);`

Then update state normally.